roles/kubernetes-apps/policy_controller/calico/templates/calico-kube-controllers.yml.j2

```diff
@@ -46,7 +46,20 @@ spec:
             requests:
               cpu: {{ calico_policy_controller_cpu_requests }}
               memory: {{ calico_policy_controller_memory_requests }}
+{% if calico_version is version('v3.3.0', '>=') %}
+          readinessProbe:
+            exec:
+              command:
+              - /usr/bin/check-status
+              - -r
+{% endif %}
           env:
+{% if calico_datastore == "kdd" and calico_version is version('v3.6.0', '>=') %}
```

Comment:
Does **kube-controllers** work for you? I had to change 'calico-kube-controllers' ClusterRole according to [calico manifest](https://docs.projectcalico.org/v3.7/manifests/calico.yaml) to make it work properly.

Reply:
I actually did forgot this part, and did not detect it in my deployment because of the cluster role got updated as a side effect of other tests.

Fixed by latest commit

Reply:
Thanks. Although, here and in the latest commit there are conditionals that prevent us from running calico+kdd for calico < v3.6 at all. So, those conditionals should be changed or it should be some formal restriction to prevent user from installing calico <v3.6 using kdd.

Reply:
In the latest commit I added the `calico-version >= 3.6` limitation because no cluster role is defined for kube-controller in previous versions in kdd mode (I looked at [3.4](https://github.com/projectcalico/calico/blob/master/_includes/v3.4/manifests/rbac.yaml) and [3.5](https://github.com/projectcalico/calico/blob/master/_includes/v3.5/manifests/rbac.yaml) templates).
I used the same reasoning for the other parts in this PR that have a constraint on `calico_version`.
So in this regard there should be no regression concerning previous versions.

Reply:
Agree. Sorry, I didn't check that. 
Seems, it would be better to exclude whole **calico-kube-controllers Deployment** from installation for "calico < v3.6 using kdd" then.

Reply:
Apparently calico kube controller seems irrelevant when kdd mode is used in versions < 3.6 so I did as you suggested (commit d988cca).

I can remove this commit if this technical choice isn't approved by everybody